mantidproject/mantid
# Update LookupRow Indexes in Model when Settings Change

**Description of work.**
Adds calls through the gui's presenter layers to inform the model that it needs to update rows. 
When:
- Exp Settings are changed, update all rows in the runs table
- a Group name is changed, update all rows in the group
- a Row's cell is changed, update that row

**To test:**
1. Load a batch, (ask me on slack for a file with some values already filled in, it's much easier than filling it all in yourself)
2. Take note of what matches should occur, (E.g: The run with an angle of 0.7 and a group name containing MAB should take the dQ/Q value from that row in the experiements settings tab)
3. Hit process on the runs tab
4. Check the right values were used. 

Fixes #33625 

*This does not require release notes* because **there are not yet any user facing changes.**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
